src/core/gui/XournalppCursor.cpp

```diff
@@ -441,12 +441,13 @@ auto XournalppCursor::createHighlighterOrPenCursor(int size, double alpha) -> Gd
     auto cursorType = control->getSettings()->getStylusCursorType();
     auto cursor = (cursorType == STYLUS_CURSOR_NONE) ? CRSR_BLANK_CURSOR : CRSR_PENORHIGHLIGHTER;
     bool bright = control->getSettings()->isHighlightPosition();
-    int height = size;
-    int width = size;
+    double cursorSize = control->getToolHandler()->getThickness() * control->getZoomControl()->getZoom();
+    int width = 2 + static_cast<int>(cursorSize);
+    int height = 2 + static_cast<int>(cursorSize);
```

Comment:
The cursor shouldn't become too big, see https://github.com/xournalpp/xournalpp/issues/4718#issuecomment-1468593480. Note that `height` and `width` are set to `90` when using the big stylus cursor or when highlighting the cursor. So what do you think about this suggestion:
```suggestion
    double cursorSize = std::min(90., control->getToolHandler()->getThickness() * control->getZoomControl()->getZoom());
    int width = static_cast<int>(std::ceil(cursorSize));
    int height = static_cast<int>(std::ceil(cursorSize));
```